src/handle.rs

```diff
@@ -1,97 +1,101 @@
 use sync::list::Node;
 use garbage::{Garbage, Bag};
-use global;
-
-
-/// Number of pinnings after which a mutator will collect some global garbage.
-const PINS_BETWEEN_COLLECT: usize = 128;
+use collector::Global;
 
 
-/// Entity that changes shared locations.
-pub struct Mutator<'scope> {
+// FIXME(stjepang): Registries are stored in a linked list because linked lists are fairly easy to
+// implement in a lock-free manner. However, traversal is rather slow due to cache misses and data
+// dependencies. We should experiment with other data structures as well.
+/// Reference to a garbage collector
+pub struct Handle {
+    /// A reference to the global data.
+    global: Arc<Global>,
     /// The local garbage objects that will be later freed.
     bag: UnsafeCell<Bag>,
-    /// This mutator's entry in the local epoch list.
-    local_epoch: &'scope Node<LocalEpoch>,
-    /// Whether the mutator is currently pinned.
+    /// This handle's entry in the local epoch list.
+    ///
+    /// It points to a node in `global`, so it is live as far as the handle has a counted reference
+    /// to `global`.
+    local_epoch: *const Node<LocalEpoch>,
+    /// Whether the handle is currently pinned.
     is_pinned: Cell<bool>,
     /// Total number of pinnings performed.
     pin_count: Cell<usize>,
 }
 
-/// An entry in the linked list of the registered mutators.
+/// An entry in the linked list of the registered handles.
 #[derive(Default, Debug)]
 pub struct LocalEpoch {
-    /// The least significant bit is set if the mutator is currently pinned. The rest of the bits
+    /// The least significant bit is set if the handle is currently pinned. The rest of the bits
     /// encode the current epoch.
     state: AtomicUsize,
 }
 
-/// A witness that the current mutator is pinned.
+/// A witness that the current handle is pinned.
 ///
-/// A reference to `Scope` is a witness that the current mutator is pinned. Lots of methods that
-/// interact with [`Atomic`]s can safely be called only while the mutator is pinned so they often
-/// require a reference to `Scope`.
+/// A `Scope` is a witness that the current handle is pinned. Lots of methods that interact with
+/// [`Atomic`]s can safely be called only while the handle is pinned so they often require a
+/// `Scope`.
 ///
 /// This data type is inherently bound to the thread that created it, therefore it does not
 /// implement `Send` nor `Sync`.
 ///
 /// [`Atomic`]: struct.Atomic.html
-#[derive(Debug)]
-pub struct Scope {
+#[derive(Clone, Copy, Debug)]
+pub struct Scope<'scope> {
+    /// A reference to the global data.
+    global: &'scope Global,
+    /// The local garbage bag.
     bag: *mut Bag, // !Send + !Sync
 }
 
 
-impl<'scope> Mutator<'scope> {
-    pub fn new() -> Self {
-        Mutator {
+impl Handle {
+    /// Number of pinnings after which a handle will collect some global garbage.
+    const PINS_BETWEEN_COLLECT: usize = 128;
+
+    pub fn new(global: &Arc<Global>) -> Self {
```

Comment:
Shouldn't this take a `Collector` instead? The fact that a `Collector` is just a newtype wrapper around a `Global` probably shouldn't be exposed to the user (and I'm not even sure that a user could construct an `Arc<Global>` since `Global` itself isn't public outside of this crate).

Reply:
Yep, looks good!